packages/react-components/react-charts-preview/library/src/components/LineChart/LineChart.tsx

```diff
@@ -816,27 +827,10 @@ export const LineChart: React.FunctionComponent<ILineChartProps> = React.forward
           </g>,
         );
       }
-      const classes = useLineChartStyles_unstable(props);
-      // Removing un wanted tooltip div from DOM, when prop not provided.
-      if (!props.showXAxisLablesTooltip) {
-        try {
-          document.getElementById(_tooltipId) && document.getElementById(_tooltipId)!.remove();
-          // eslint-disable-next-line no-empty
-        } catch (e) {}
-      }
       // Used to display tooltip at x axis labels.
       if (!props.wrapXAxisLables && props.showXAxisLablesTooltip) {
         const xAxisElement = d3Select(xElement).call(_xAxisScale);
-        try {
-          document.getElementById(_tooltipId) && document.getElementById(_tooltipId)!.remove();
-          // eslint-disable-next-line no-empty
-        } catch (e) {}
-        const tooltipProps = {
-          tooltipCls: classes.tooltip!,
-          id: _tooltipId,
-          xAxis: xAxisElement,
-        };
-        xAxisElement && tooltipOfXAxislabels(tooltipProps);
+        setXAxisElement(xAxisElement.node());
```

Comment:
instead of setting xAxisElement can you set the tooltipprops here directly.

Reply:
tooltip props now has only these 2 elements out of which tooltipRef is not a state variable, thus need not be set. const tooltipProps = {
        xAxis: xAxis,
        div: tooltipRef.current,
      };